modules/packages/LinearAlgebra.chpl

```diff
@@ -1313,6 +1313,50 @@ proc lu (A: [?Adom] ?eltType) {
   return (LU,ipiv);
 }
 
+private proc getQ(A, tau) {
+  var n = A.domain.shape[0];
+  var I = eye(A.domain);
+  var Q = I;
+  for i in 1..n {
+    var v: [1..n] real;
```

Comment:
It looks like `proc Vector` overload used here creates a new array from the argument. So, we don't have to create a new array at each iteration. Could you put this right before the loop (and make sure that it doesn't break anything) ?